File: app/src/main/java/com/selepdf/hackovid/model/Product.java
package com.selepdf.hackovid.model;

public class Product extends GeneralItem {

    private Category category;
    private Store store;

    public Product(String name, String description, String thumbnail, float price, float rating, String[] images, Category category, Store store) {
        super(name, description, thumbnail, price, rating, images);
        this.category = category;
        this.store = store;
    }

    public Product(String name, String description, String price) {
        super(name, description, price);
    }

    public Category getCategory() {
        return category;
    }

    public void setCategory(Category category) {
        this.category = category;
    }

    public Store getStore() {
        return store;
    }

    public void setStore(Store store) {
        this.store = store;
    }
}

File: hackovid_be/model/Product.js
var mongoose = require('mongoose');

var Store = require('./Store').Schema;

var ProductSchema = new mongoose.Schema({  
  name: String,
  images: [String],
  category: { type: mongoose.Schema.ObjectId, ref: "ProductCategory" },
  description: String,
  store: Store,
  price: Number,
  rating: Number,
});
mongoose.model('Product', ProductSchema);

module.exports = mongoose.model('Product');